extensions/azurecore/src/account-provider/utils/fileEncryptionHelper.ts

```diff
@@ -97,13 +97,15 @@ export class FileEncryptionHelper {
 			return `${decipherIv.update(plaintext, this._binaryEncoding, 'utf8')}${decipherIv.final('utf8')}`;
 		} catch (ex) {
 			Logger.error(`FileEncryptionHelper: Error occurred when decrypting data, IV/KEY will be reset: ${ex}`);
-			// Reset IV/Keys if crypto cannot encrypt/decrypt data.
-			// This could be a possible case of corruption of expected iv/key combination
-			await this.deleteEncryptionKey(this._ivCredId);
-			await this.deleteEncryptionKey(this._keyCredId);
-			this._ivBuffer = undefined;
-			this._keyBuffer = undefined;
-			await this.init();
+			if (resetOnError) {
+				// Reset IV/Keys if crypto cannot encrypt/decrypt data.
+				// This could be a possible case of corruption of expected iv/key combination
+				await this.deleteEncryptionKey(this._ivCredId);
+				await this.deleteEncryptionKey(this._keyCredId);
+				this._ivBuffer = undefined;
+				this._keyBuffer = undefined;
```

Comment:
What's the difference between a IV buffer and key buffer? The terminology is unclear to me.

Reply:
They are related to AES encryption algorithm, IV: initialization vector and Key: Encryption Key

You can read more here: https://learn.microsoft.com/en-us/dotnet/standard/security/generating-keys-for-encryption-and-decryption#symmetric-keys